infrastructure/templates/public-api/components/functionApp.bicep

```diff
@@ -49,6 +49,13 @@ param userAssignedManagedIdentityParams {
   principalId: string
 }?
 
+@description('An existing App Registration registered with Entra ID that will be used to control access to this Function App')
+param entraIdAuthentication {
+  appRegistrationClientId: string
+  allowedClientIds: string[]
+  allowedPrincipalIds: string[]
+}?
```

Comment:
I was wondering why we can't create the app registration in this template. Is it because it's not supported in Bicep or because our deployment principal has insufficient role permissions?

Reply:
Haha.  Sooo...

ARM and Bicep operate on resources in Subscriptions and Resource Groups, but App Registrations are part of Entra ID, not ARM resources.

The only way to achieve this in ARM / Bicep is to use the "DeploymentScript" resource type, and then use it to run Azure CLI scripts and capture the outputs.  This needs a user-assigned identity creating who is privileged enough to do the actions that you're asking them to do.

Looking into it, there was no way that we'd get permission to have our SPNs be able to create App Registrations against DfE's Entra ID (especially for Prod and Pre-Prod, it's Service Tickets only), so I went down this approach.

Reply:
Ok cool, I understand the separation now, thanks.